modules/versions.xqm

```diff
@@ -120,48 +102,33 @@ function versions:find-version($packages as element(package)*, $minVersion as xs
             versions:find-version(tail($packages), $minVersion, $maxVersion, $newer)
 };
 
-declare 
-    %private 
-function versions:is-newer($available as xs:string, $installed as xs:string) as xs:boolean {
-    let $verInstalled := tokenize($installed, "\.")
-    let $verAvailable := tokenize($available, "\.")
-    return
-        versions:compare-versions(
-            $verInstalled, 
-            $verAvailable, 
-            function($version1, $version2) {
-                number($version1) >= number($version2)
-            }
-        )
-};
-
-declare 
-    %private 
-function versions:is-older($available as xs:string, $installed as xs:string) as xs:boolean {
-    let $verInstalled := tokenize($installed, "\.")
-    let $verAvailable := tokenize($available, "\.")
-    return
-        versions:compare-versions(
-            $verInstalled, 
-            $verAvailable, 
-            function($version1, $version2) {
-                number($version1) <= number($version2)
-            }
-        )
-};
-
-declare 
-    %private 
-function versions:compare-versions($installed as xs:string*, $available as xs:string*, $compare as function(*)) as xs:boolean {
-    if (empty($installed)) then
-        exists($available)
-    else if (empty($available)) then
-        false()
-    else if (head($available) = head($installed)) then
-        if (count($available) = 1 and count($installed) = 1) then
+(:~
+ : Find packages whose eXist version requirements meet the client's eXist version
+ : 
+ : For example, via app.xqm or list.xq, a client may request the subset of a package's
+ : releases that are compatible with eXist 5.3.0. The function examines each release's
+ : eXist dependency declarations (if present) and returns all matching packages.
+ :)
+declare function versions:find-packages-satisfying-exist-version-requirements(
+    $packages as element(package)*,
+    $exist-version-semver as xs:string, 
+    $min-version as xs:string?, 
+    $max-version as xs:string?
+) as element(package)* {
+    for $package in $packages
+    let $satisfies-semver-min-requirement := 
+        if (exists($package/requires/@semver-min)) then
+            semver:ge($exist-version-semver, $package/requires/@semver-min, true())
+        else
             true()
+    let $satisfies-semver-max-requirement := 
+        if (exists($package/requires/@semver-max)) then
+            semver:lt($exist-version-semver, $package/requires/@semver-max, true())
         else
-            versions:compare-versions(tail($installed), tail($available), $compare)
-    else
-        $compare(head($available), head($installed))
-};
+            true()
+    return
+        if ($satisfies-semver-min-requirement and $satisfies-semver-max-requirement) then
+            $package
+        else
+            ()
+};
```

Comment:
let's make git happy and add a new line here :)

Reply:
Fixed. Thanks!